pkg/operator/starter.go

```diff
@@ -114,6 +118,24 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 	kubeInformersNamespaced.Start(ctx.Done())
 	kubeInformersForNamespaces.Start(ctx.Done())
 
+	// Poll every minute to ensure removal of the 4.4 deployment in case of downgrade
+	// to minimize resource usage and the opportunity for contention.
+	deployClient := kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+	deployName := "service-ca"
+	go wait.Until(func() {
+		_, err := deployClient.Get(deployName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return
+		}
+		if err != nil {
+			klog.Warningf("Failed to retrieve 4.4 deployment: %v", err)
+		}
+		err = deployClient.Delete(deployName, &metav1.DeleteOptions{})
```

Comment:
why not delete directly, why a GET?

Shouldn't we stop the loop as soon as it succeeds?

Reply:
Done.

Reply:
I think the original code is still in place, did you add your changes to your latest commit?

Reply:
Get is still there. But I don't care too much.

Reply:
But I would prefer to see the loop stop when the deployment is gone.

Reply:
I reverted the changes as per @deads2k. iirc his assertion is that continual reconciliation is required to avoid confusing users when behavior is inconsistent (removes a deployment on startup not subsequently).

As to why get, I thought it would be preferable to see 'get' in the log vs 'delete' if continually reconciling but maybe that's specious logic?

Reply:
ok, fine with me.

Reply:
Am still confused why the deployment is different from the other resources in `cleanupUnifiedDeployment`.

Reply:
That's what I asked in https://github.com/openshift/service-ca-operator/pull/93#discussion_r382408155. Those other resources don't have an operational impact for the operator, so deleting them isn't so critical, but the principle of consistent behavior suggests continual reconciliation for them too. 

I guess I've answered my own question. Maybe reconcile on a longer timeline - 20m?

Reply:
Fair enough. 20m is also fine, so is 1m.